core/kotlinx-coroutines-core/src/main/kotlin/kotlinx/coroutines/experimental/channels/Subscribables.kt

```diff
@@ -0,0 +1,36 @@
+package kotlinx.coroutines.experimental.channels
+
+import kotlin.properties.ReadOnlyProperty
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+
+public interface Subscribable<out T> {
+  public fun openSubscription(): SubscriptionReceiveChannel<T>
+}
+
+public interface SubscribableValue<out T> : Subscribable<T>, ReadOnlyProperty<Any, T> {
+
+  /**
+   * Current value if any.
+   *
+   * Access this property may throws [IllegalStateException] if there is no value set yet.
+   *
+   * Use [valueOrNull] to get `null` when the value is not set or `openSubscription().consume { receive() }` to suspend until a value is available.
+   */
+  public val value: T
+
+  public val valueOrNull: T?
```

Comment:
We should support `lateinit`?

Is `Deferred<SubscribableValue>` a good replacement?

Reply:
I think yes, because this is the actual current behavior of `ConflatedBroadcastChannel`.

https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.experimental.channels/-conflated-broadcast-channel/value.html

Reply:
Ok, discard this point.